packages/bubble-dev/start-preset/src/build.ts

```diff
@@ -87,36 +88,54 @@ export const buildReactNative = async (dir: string): Promise<StartPlugin<{}, {}>
 
 export const buildNode = async (dir: string): Promise<StartPlugin<{}, {}>> => {
   const { babelConfigNodeBuild } = await import('@bubble-dev/babel-config')
-
-  return sequence(
-    env({ BABEL_ENV: 'production' }),
-    find([
+  const { default: globby } = await import('globby')
+  const allFiles = await globby(
+    [
       `${dir}/src/**/*.{js,jsx,ts,tsx}`,
       `!${dir}/src/**/*.{native,ios,android}.{js,jsx,ts,tsx}`,
       `!${dir}/src/**/*.d.ts`,
-    ]),
-    read,
-    babel(babelConfigNodeBuild),
-    rename((file) => file.replace(/\.(ts|tsx|jsx)$/, '.js')),
-    write(`${dir}/build/node/`),
-    find(`${dir}/src/**/*.json`),
-    copy(`${dir}/build/node/`),
-    plugin('test', () => async () => {
-      const path = await import('path')
-      const { access } = await import('pifs')
-      const fullPath = path.resolve(`${dir}/build/node/index.js`)
-      let hasFile = false
-
-      try {
-        await access(fullPath)
-        hasFile = true
-      } catch {}
-
-      if (hasFile) {
-        await import(fullPath)
-      }
-    })
+    ],
+    {
+      ignore: ['node_modules/**'],
+      deep: Infinity,
+      onlyFiles: true,
+    }
   )
+  const extRegExp = /\.(js|jsx|ts|tsx)$/
+  const nodeFiles = allFiles.filter((file) => {
+    if (allFiles.includes(file.replace(extRegExp, '.node.$1'))) {
+      return false
```

Comment:
shouldnt we return `true`?
like if we do find `potato.node.js` we want to use it instead of `potato.js`?

Reply:
This part is tricky, I also spend some time to figure out what is happening here 😅
The confusing part is the `file.replace()` that grabs the file extension `.(js|jsx|ts|tsx)` from currently filtered file, changes is for `node.(js|jsx|ts|tsx)` and checks if file of the same name, but with new extension is already in `allFiles`. If there already is a file like that in `allFiles` we have to return `false`, because the currently filtered file is the one without `node` extension.

I figured it out on an example. Let's say we have 3 files in `allFiles`: "index.jsx", "Root.jsx" & "Root.node.jsx". We start by grabbing "index.jsx" and in this `if` we check if `allFiles` includes a "index.node.jsx" file, it doesn't, so we skip `if` and go to `true`.
Then we grab "Root.jsx" and in the `if` we check if there is maybe a "Root.node.jsx" in `allFiles` already. There is, so we go inside `if` and return `false` for "Root.jsx".
Then we grab last file "Root.node.jsx" and check if there is maybe a "Root.node.node.jsx" file, there isn't so we skip the `if` and return `true`.

Reply:
<img width="549" alt="Screenshot 2021-01-25 at 13 49 19" src="https://user-images.githubusercontent.com/16437281/105708038-259d9100-5f14-11eb-95bc-f19cec76b646.png">